### PR TITLE
Expanding unit tests to include L2 basis with MapType INTEGRAL [l2-integral-dev]

### PR DESCRIPTION
**Pull request by mlstowell**
The goal of this PR is to expand the unit tests to identify and remove any inconsistencies that may exist in the implementation of BilinearForm and LinearForm integrators related to the two map types supported by the L2 basis functions.  Some of these issues have already been addressed in other PRs, notably PR #1931. 

This PR was created in response to issue #1037.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1995](https://github.com/mfem/mfem/pull/1995) | @mlstowell | @v-dobrev | @barker29 + @kmittal2 | 01/25/21 | 03/02/21 | 03/16/21 | |

### CODE REVIEW

File: fem/bilininteg.cpp [hunk 1]
```diff
@@ -907,9 +907,10 @@ void MassIntegrator::AssembleElementMatrix
    for (int i = 0; i < ir->GetNPoints(); i++)
    {
       const IntegrationPoint &ip = ir->IntPoint(i);
-      el.CalcShape(ip, shape);
-
       Trans.SetIntPoint (&ip);
+
+      el.CalcPhysShape(Trans, shape);
+
```
Comment (tzanio) [inline fem/bilininteg.cpp:913]: Is this change really necessary? Aren't the reference and physical values the same?

Comment (mlstowell) [reply fem/bilininteg.cpp:913]: Not if the `MapType == INTEGRAL`.

Comment (tzanio) [reply fem/bilininteg.cpp:913]: Ah, true

**Review by barker29 — APPROVED** [review]
I appreciate that a one-line code change comes with hundreds of lines of unit tests. This looks good.

**Review by kmittal2 — APPROVED** [review]
Looks good to me. During the review, I found that the styling was off at a bunch of places (even before this PR) so I have fixed that.

**Comment by mlstowell** [comment]
Just for the record.  @kmittal2 pointed out the style issues to me and generously offered to fix them.  I fully support and appreciate his additions to this PR. Thanks again, @kmittal2!

**Comment by kmittal2** [comment]
> Just for the record. @kmittal2 pointed out the style issues to me and generously offered to fix them. I fully support and appreciate his additions to this PR. Thanks again, @kmittal2!

Happy to help @mlstowell.

**Review by tzanio — APPROVED** [review]

**Comment by tzanio** [comment]
@v-dobrev, any objections to merge this in `next`?

**Comment by v-dobrev** [comment]
Merged in `next` for testing...
